core/src/main/java/com/expediagroup/streamplatform/streamregistry/core/services/ProcessService.java

```diff
@@ -56,9 +60,48 @@
     }
     processValidator.validateForCreate(process);
     process.setSpecification(handlerService.handleInsert(process));
+    process.getZones().forEach(zoneKey ->
+      process.getInputs().forEach(input -> createConsumer(
+        new Consumer(
+          new ConsumerKey(
+            input.getStream().getDomainKey().getName(),
+            input.getStream().getName(),
+            input.getStream().getVersion(),
+            zoneKey.getName(),
+            process.getKey().getName()
+          ),
+          process.getSpecification(),
+          process.getStatus()
+        )
+      )));
+
+    process.getZones().forEach(zoneKey ->
+      process.getOutputs().forEach(output -> createProducer(
+        new Producer(
+          new ProducerKey(
+            output.getStream().getDomainKey().getName(),
+            output.getStream().getName(),
+            output.getStream().getVersion(),
+            zoneKey.getName(),
+            process.getKey().getName()
+          ),
+          process.getSpecification(),
+          process.getStatus()
+        )
+      )));
     return save(process);
   }
 
+  @PreAuthorize("hasPermission(#consumer, 'CREATE')")
+  public void createConsumer(Consumer consumer) {
```

Comment:
Can we rename these methods something like `testCreateConsumer` etc, to make it clear that the intention is merely to test the permission, not to actually create the consumer/producer.

Then move these methods inside of the respective services they belong to.

Reply:
updated